packages/flutter/lib/src/gestures/selection_recognizers.dart

```diff
@@ -0,0 +1,714 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart' show HardwareKeyboard, LogicalKeyboardKey;
+
+import 'constants.dart';
+import 'drag_details.dart';
+import 'events.dart';
+import 'monodrag.dart' show GestureDragCancelCallback;
+import 'recognizer.dart';
+import 'tap.dart' show GestureTapCallback, GestureTapCancelCallback, GestureTapDownCallback, GestureTapUpCallback, TapDownDetails, TapUpDetails;
+
+enum _GestureState {
+  ready,
+  possible,
+  accepted,
+}
+
+/// {@macro flutter.gestures.tap.GestureTapDownCallback}
+///
+/// The consecutive tap count at the time the pointer contacted the screen is given by [TapStatus.consecutiveTapCount].
+///
+/// Used by [TapAndDragGestureRecognizer.onTapDown].
+typedef GestureTapDownWithTapStatusCallback  = void Function(TapDownDetails details, TapStatus status);
+
+/// {@macro flutter.gestures.tap.GestureTapUpCallback}
+///
+/// The consecutive tap count at the time the pointer contacted the screen is given by [TapStatus.consecutiveTapCount].
+///
+/// Used by [TapAndDragGestureRecognizer.onTapUp].
+typedef GestureTapUpWithTapStatusCallback  = void Function(TapUpDetails details, TapStatus status);
+
+/// {@macro flutter.gestures.dragdetails.GestureDragStartCallback}
+///
+/// The consecutive tap count when the drag was initiated is given by [TapStatus.consecutiveTapCount].
+///
+/// Used by [TapAndDragGestureRecognizer.onStart].
+typedef GestureDragStartWithTapStatusCallback = void Function(DragStartDetails details, TapStatus status);
+
+/// {@macro flutter.gestures.dragdetails.GestureDragUpdateCallback}
+///
+/// The consecutive tap count when the drag was initiated is given by [TapStatus.consecutiveTapCount].
+///
+/// Used by [TapAndDragGestureRecognizer.onUpdate].
+typedef GestureDragUpdateWithTapStatusCallback = void Function(DragUpdateDetails details, TapStatus status);
```

Comment:
Since the TapStatus is specific to selection gesture, consider create a special DragUpdateDetails? I am not sure what is long term plan for these detail class @dkwingsmt may have better suggestion.